src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchClient.kt

```diff
@@ -0,0 +1,68 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+@Component
+class PrisonerSearchClient(private val prisonerSearchWebClient: WebClient) {
+  fun matchPrisoners(prisonerSearchRequest: PrisonerSearchRequest): MatchPrisonersResponse {
+    return prisonerSearchWebClient
+      .post()
+      .uri("/match-prisoners")
+      .accept(MediaType.APPLICATION_JSON)
+      .bodyValue(
+        mapOf(
+          "nomsNumber" to prisonerSearchRequest.prisonNumber,
+          "firstName" to prisonerSearchRequest.firstName,
+          "lastName" to prisonerSearchRequest.lastName,
+          "dateOfBirth" to prisonerSearchRequest.dob?.format(DateTimeFormatter.BASIC_ISO_DATE)
+        )
+      )
+      .retrieve()
+      .bodyToMono<MatchPrisonersResponse>()
+      .block()!!
+  }
+}
+
+data class MatchPrisonersResponse(
+  val matches: List<PrisonerMatch>,
+  val matchedBy: MatchedBy
+)
+
+data class PrisonerMatch(
+  val prisoner: Prisoner
+)
+
+data class Prisoner(
```

Comment:
This is a basic version of the `Prisoner` record as returned by the API with a subset of fields. For now I've included just the fields I think we'll be needing, but we can always add to it. At this point I just wanted to be able to do the simplest thing in order to call the API

Reply:
Sounds good. I wonder if longer term we should generate the API model from the OpenAPI docs?

Reply:
I think you are 100% right there - once we get this basic idea working, then yes, definitely we should generate the classes from the swagger spec 👍